doc/sphinx/addendum/generalized-rewriting.rst

```diff
@@ -759,10 +759,49 @@ example of a mostly user-space extension of the algorithm.
 Constant unfolding
 ~~~~~~~~~~~~~~~~~~
 
-The resolution tactic is based on typeclasses and hence regards user-defined
-:term:`constants <constant>` as transparent by default. This may slow down the
-resolution due to a lot of unifications (all the declared ``Proper``
-instances are tried at each node of the search tree). To speed it up,
+.. cmd::  Hint {| Variables | Constants } {| Opaque | Transparent } : rewrite.
+
+   By default, :tacn:`setoid_rewrite` and :tacn:`rewrite_strat` unfold global
+   :term:`definitions <constant>` during rewrite-rule matching, but do not
+   unfold local definitions. The former may slow down matching, whereas the
+   latter may cause matches to be missed. This behavior is configurable using
+   transparency hints in the special hint database ``rewrite`` as shown,
```

Comment:
The `rewrite` and `setoid_rewrite` and `rewrite_strat` tactics are hardwired to use the database called `rewrite`.  It is special because its name is hardcoded in the ML source code, nothing else.

Reply:
Indeed, I think it is important to communicate that this rewrite database behaves differently from user-created databases.

Reply:
Yes.  But just saying "special" leaves the reader wondering just how it is special.  And IIUC that it doesn't work for other hint databases.

Reply:
> this rewrite database behaves differently from user-created databases.

It's not the database that behaves differently, though, right?  AIUI, there are three "special" databases:
- core - the default db for (e)auto, trivial, etc
- typeclass_instances - the default db for typeclasses eauto
- rewrite - the default db for transparency information for rewrite & setoid_rewrite & rewrite_strat (and autorewrite?), which (unfortunately?) do not allow choosing a different database for transparency information

Reply:
@jfehrle what is "it" in "IIUC it doesn't work for other hint databases."?  I believe these commands do work and are meaningful for other hint databases (they should control unification / matching heuristics in (e)auto), it is just that the rewrite database is the one that gets picked up by the rewrite tactics